app/redux/reducers/SettingsReducer.js

```diff
@@ -38,8 +62,46 @@ const SettingsReducer = (state = defaultState, action) => {
         case DISMISS_DEFAULT_METRIC:
             return Object.assign({}, state, {
                 isEditingDefaultMetric: false,
-            });                        
-        case SAVE_END_SET_TIMER:
+            });
+        case PRESENT_COLLAPSED_METRICS: 
+            changes = { currentMetricPosition: action.metricPosition }
+                if (action.quantifier === 'Average' || action.quantifier === 'Abs Loss') {
+                    changes.isEditingAvgMetric = true;
+                } else if (action.quantifier === 'Best Ever') {
+                    changes.isEditingBestEverMetric = true;
+                } else {
+                    changes.isEditingMetric = true;
+                }
+            return Object.assign({}, state, changes);
+        case DISMISS_COLLAPSED_METRICS: 
+            return Object.assign({}, state, {
+                isEditingMetric: false,
+                isEditingAvgMetric: false,
+                isEditingBestEverMetric: false,
+                currentMetricPosition: null
+            });
+        case PRESENT_QUANTIFIERS: 
+            return Object.assign({}, state, {
+                isEditingQuantifier: true,
+                currentQuantifierPosition: action.quantifierPosition,
+                currentMetricPosition: action.metricPosition
+            });
+        case DISMISS_QUANTIFIERS: 
+            return Object.assign({}, state, {
+                isEditingQuantifier: false,
+                currentMetricPosition: null,
+                currentQuantifierPosition: null
+            });
+        case SAVE_COLLAPSED_METRIC:
+            changes = {}
+            changes[action.position] = action.metric
+            return Object.assign({}, state, changes);
+        case SAVE_QUANTIFIER:
+            changes = {}
+            changes[action.position] = action.quantifier
+            changes[state.currentMetricPosition] = 'Velocity'
```

Comment:
- [ ] Why is this defaulted to Velocity? Either it's an issue or it needs a comment

Reply:
TBD - Matt